GliaWidgets/ViewController/Survey/Survey.ViewController.Props.swift

```diff
@@ -0,0 +1,243 @@
+        sdkQuestion: CoreSdkClient.Survey.Question,
+        getQuestion: @escaping (String) -> SurveyQuestionPropsProtocol?,
+        getQuestions: @escaping () -> [SurveyQuestionPropsProtocol],
+        setQuestion: @escaping (String, SurveyQuestionPropsProtocol) -> Void,
+        updateProps: @escaping (Self) -> Void
+    ) -> Survey.ScaleQuestionView.Props {
+
+        var scaleProps = Survey.ScaleQuestionView.Props(
+            id: sdkQuestion.id.rawValue,
+            title: sdkQuestion.text,
+            isRequired: sdkQuestion.required
+        )
+        let handleScaleOptionSelection = { (option: Survey.Option<Int>) in
+            guard var optionProps = getQuestion(sdkQuestion.id.rawValue) as? Survey.ScaleQuestionView.Props else { return }
+            optionProps.selected = option
+            optionProps.showValidationError = false
+            optionProps.answerContainer = .int(option.value)
+            setQuestion(sdkQuestion.id.rawValue, optionProps)
+            var newViewControllerProps = props()
+            newViewControllerProps.questionsProps = getQuestions()
+            updateProps(newViewControllerProps)
+        }
+        scaleProps.options = [
+            .init(name: "1", value: 1, select: handleScaleOptionSelection),
+            .init(name: "2", value: 2, select: handleScaleOptionSelection),
+            .init(name: "3", value: 3, select: handleScaleOptionSelection),
+            .init(name: "4", value: 4, select: handleScaleOptionSelection),
+            .init(name: "5", value: 5, select: handleScaleOptionSelection)
+        ]
+
+        return scaleProps
+    }
+
+    static func makeBooleanProps(
+        props: @escaping () -> Self,
+        sdkQuestion: CoreSdkClient.Survey.Question,
+        getQuestion: @escaping (String) -> SurveyQuestionPropsProtocol?,
+        getQuestions: @escaping () -> [SurveyQuestionPropsProtocol],
+        setQuestion: @escaping (String, SurveyQuestionPropsProtocol) -> Void,
+        updateProps: @escaping (Self) -> Void
+    ) -> Survey.BooleanQuestionView.Props {
+
+        var booleanProps = Survey.BooleanQuestionView.Props(
+            id: sdkQuestion.id.rawValue,
+            title: sdkQuestion.text,
+            isRequired: sdkQuestion.required
+        )
+        let handleBooleanOptionSelection = { (option: Survey.Option<Bool>) in
+            guard var optionProps = getQuestion(sdkQuestion.id.rawValue) as? Survey.BooleanQuestionView.Props else { return }
+            optionProps.selected = option
+            optionProps.showValidationError = false
+            optionProps.answerContainer = .boolean(option.value)
+            setQuestion(sdkQuestion.id.rawValue, optionProps)
+            var newViewControllerProps = props()
+            newViewControllerProps.questionsProps = getQuestions()
+            updateProps(newViewControllerProps)
+        }
+        booleanProps.options = [
+            .init(name: "Yes", value: true, select: handleBooleanOptionSelection),
```

Comment:
These will need to be localized as well.

Reply:
Yes, I'll move this string in the next PR together with theme stuff.